File: movies/python/movies_server.py
import asyncio

import grpc
import grpc.experimental.aio
from movies_pb2 import AllMoviesResponse, MovieEntry
from movies_pb2_grpc import AllMoviesServicer, add_AllMoviesServicer_to_server


class AsyncMoviesService(AllMoviesServicer):
    async def get_all_movies(self, request, context):
        all_movies = [
            MovieEntry(title="Movie title #1")
        ]

        return AllMoviesResponse(movies=all_movies)


async def main():
    server_port = 50051

    grpc.experimental.aio.init_grpc_aio()

    # Create the async gRPC server
    server = grpc.experimental.aio.server()
    server.add_insecure_port(f"[::]:{server_port}")
    print(f"Server listening on: 0.0.0.0:{server_port}")

    add_AllMoviesServicer_to_server(AsyncMoviesService(), server)

    await server.start()
    await server.wait_for_termination()


if __name__ == "__main__":
    asyncio.run(main())

File: movies/javascript/movies_server.js
const grpc = require('@grpc/grpc-js');
const protoLoader = require('@grpc/proto-loader');

const packageDefinition = protoLoader.loadSync('../../protobufs/movies.proto', {});
const moviesPackage = grpc.loadPackageDefinition(packageDefinition);


function get_all_movies(call, callback) {
  let all_movies = [
    { title: "From Javascript" }
  ]

  return callback(null, {movies: all_movies});
}

// Create a new gRPC server
const server = new grpc.Server();

server.addService(
  moviesPackage.AllMovies.service,
  {
    get_all_movies: get_all_movies
  }
);

// Run the gRPC server async and insecurely (without auth or encryption) using the default gRPC port of 50051
server.bindAsync('0.0.0.0:50051', grpc.ServerCredentials.createInsecure(), () => {
  server.start();
});
